integration-tests/shard-operator-integration-tests/shard-operator-integration-tests-v2/src/test/java/com/redhat/service/smartevents/shard/operator/v2/cucumber/steps/KubernetesSteps.java

```diff
@@ -48,4 +51,34 @@ public void createSecret(String name, DataTable values) {
                 .build();
         context.getClient().secrets().inNamespace(context.getNamespace()).withName(name).createOrReplace(secret);
     }
+
+    @Then("^the Deployment \"([^\"]*)\" is ready within (\\d+) (?:minute|minutes)$")
+    public void theDeploymentIsInStateWithinMinutes(String name, int timeoutInMinutes) throws TimeoutException {
+        Awaitility.await()
+                .atMost(Duration.ofMinutes(timeoutInMinutes))
+                .pollInterval(Duration.ofSeconds(5))
+                .until(
+                        () -> {
+                            Deployment deployment = context.getClient().apps().deployments().inNamespace(context.getNamespace()).withName(name).get();
+                            if (deployment == null) {
+                                return false;
+                            }
+                            //I am not sure if this code is valid for v2: deployment.getStatus().getConditions().stream().anyMatch(d -> d.getType().equals("Available") && d.getStatus().equals("True"))
+                            //Th PR of MGDOBR-1248 contains this implementation of the `isReadyV2` function in CustomResourceStatus: conditions.stream().allMatch(c -> ConditionStatus.True.equals(c.getStatus()))
+                            //return deployment.getStatus().getConditions().stream().allMatch(d -> d.getStatus().equals("True"));
+                            return deployment.getStatus().getConditions().stream().anyMatch(d -> d.getType().equals("Available") && d.getStatus().equals("True"));
```

Comment:
The code for conditions has changed in our own custom resources, but it shouldn't affect the availability of a Kubernetes Deployment which is a built in resource

Reply:
Remove the comment when you have figured out the correct way to check if the resource is available